### PR TITLE
Implement Sign & Decrypt in PKCS11

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: include/aos/common/crypto.hpp [hunk 1]
```diff
@@ -287,6 +345,26 @@ class ECDSAPrivateKey : public PrivateKeyItf {
...
+    /**
+     * Calculates a signature of a given digest.
+     *
+     * @param allocator tmp objects allocator.
+     * @param digest input hash digest.
+     * @param options signing options.
+     * @param[out] signature result signature.
+     * @return Error.
+     */
+    Error Sign(Allocator& allocator, const Array<uint8_t>& digest, const SignOptions& options,
```
Comment (al1img) [inline include/aos/common/crypto.hpp:381]: Could we sign without allocator?

File: tests/iam/certhandler/mocks/privatekey.hpp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+#include "aos/common/crypto.hpp"
+
+namespace aos {
+namespace crypto {
+
+/**
+ * RSA private key.
+ */
+class RSAPrivateKeyStub : public PrivateKeyItf {
```
Comment (al1img) [inline tests/iam/certhandler/mocks/privatekey.hpp:19]: Why do you need different KeyTypes? I guess one `PrivateKeyMock` would be enough, wouldn't be?

File: include/aos/common/crypto.hpp [hunk 3]
```diff
@@ -351,6 +429,21 @@ inline Error EncodeBigInt(const Array<uint8_t>& number, Array<uint8_t>& asn1Valu
...
 }
 
+/**
+ * Creates ASN1 sequence from already encoded DER items.
+ *
+ * @param items DER encoded items.
+ * @param[out] asn1Value result ASN1 value.
+ * @result Error.
+ */
+inline Error EncodeDERSequence(const Array<Array<uint8_t>>& items, Array<uint8_t>& asn1Value)
```
[the remark this answers is not in the document]
Comment (al1img) [reply include/aos/common/crypto.hpp:365]: Where it will be implemented then?

[the remark this answers is not in the document]
Comment (mykola-kobets-epam) [reply include/aos/common/crypto.hpp]: Modification of the provider interface involved a lot of changes not related to the implementation. So in order to make a review process easier I separated it into a separate commit.

**Review by al1img — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

**Pull request merged**